soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -344,6 +344,26 @@ DT_FOREACH_STATUS_OKAY(nxp_lpc_ctimer, CTIMER_CLOCK_SETUP)
 	POWER_DisablePD(kPDRUNCFG_PD_LDOGPADC);
 #endif
 #endif
+#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(dac0), nxp_lpdac, okay)
+#if defined(CONFIG_SOC_LPC55S36)
```

Comment:
I'd prefer we don't duplicate the init code for the VREF here- instead, combine the conditional for the LPDAC and LPADC to set up the VREF if either peripheral is enabled